### DISCUSSION TITLE
[HELP] Use Function Keys to Open Safari Application?

### DISCUSSION THREAD

**Discussion by JohnCreley** [pqrs-org/Karabiner-Elements#3485] [Q&A]
Hello users/programmers, I have started using Karabiner-Elements, & I had one question about the certain command "al_internet_browser": I am struggling to get the command to open my Safari application on my Ventura OS 13 iMac. [I went into detail about how I set up the custom keystroke to take effect, & how I also troubleshooted the cause](https://imgur.com/gallery/jkmiGCb) & how I was eventually unsuccessful in doing so.
     I want to be able to open the Safari application by pressing f5 using Karabiner-Elements, as well as other applications (Word, Excel, etc. on other function keys), but am quite stumped how. I don't have adequate Java skills, but I still did my best to create a .json file & [requested a pull on GitHub in the hopes that Karabiner-Elements's team approves it](https://github.com/pqrs-org/KE-complex_modifications/pull/1385) & I can import from the internet, but in the event that it is unsuccessful I'd like help from this community on what to do.
     It seems as though in previous editions it was possible, but it is no longer so. Thanks for any input!

**Comment by evan-liu** [comment]
I think this should work: 

```json
{
  "type": "basic",
  "from": {
    "key_code": "f5"
  },
  "to": [
    {
      "shell_command": "open -a Safari.app"
    }
  ]
}
```

Or [`map('f5').toApp('Safari')`](https://stackblitz.com/edit/karabiner-ts-open-app?file=rules.js) with [karabiner.ts](https://github.com/evan-liu/karabiner.ts).
